### PR TITLE
adding recipe and patch for wannier90-3.1.0 with toolchain gofb

### PR DESCRIPTION
**Pull request by MoiseRousseau**
All in the title. For a client who wants to build ABINIT using gcc toolchain

### CODE REVIEW

File: easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gfort.patch [hunk 1]
```diff
@@ -0,0 +1,24 @@
+--- config/make.inc.gfort.orig	2020-03-05 13:41:10.000000000 -0500
++++ config/make.inc.gfort	2022-06-02 09:22:27.141245863 -0400
+@@ -4,11 +4,11 @@
+ F90 = gfortran
+ 
+ #Add next two lines for parallel postw90. Choose the correct name of the mpi f90 wrapper
+-#COMMS  = mpi
+-#MPIF90 = mpgfortran #mpif90
++COMMS  = mpi
++MPIF90 = mpif90
```
Comment (mboisson) [inline easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gfort.patch]: Use `$(MPIF90)` to make the patch more generic (i.e. take the value from the environment variable `MPIF90`

File: easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gfort.patch [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ 
+-FCOPTS = -O3
+-LDOPTS =
++FCOPTS = -O2
++LDOPTS = -O2
```
Comment (mboisson) [inline easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gfort.patch]: why change the optimisation values ?

File: easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gfort.patch [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ 
+ #Next two lines are good for debugging
+ #FCOPTS = -fstrict-aliasing  -fno-omit-frame-pointer -fno-realloc-lhs -fcheck=bounds,do,recursion,pointer -ffree-form -Wall -Waliasing -Wsurprising -Wline-truncation -Wno-tabs -Wno-uninitialized -Wno-unused-dummy-argument -Wno-unused -Wno-character-truncation -O1 -g -fbacktrace
+@@ -34,4 +34,4 @@
+ #   sudo apt-get install libblas-dev liblapack-dev
+ # on Ubuntu
+ #=======================
+-LIBS = -llapack -lblas
++LIBS = -lflexiblas
```
Comment (mboisson) [inline easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gfort.patch]: Use `$(LIBLAPACK) $(LIBBLAS)` to be more generic

File: easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gofb-2020a.eb [hunk 4]
```diff
@@ -0,0 +1,35 @@
+easyblock = 'ConfigureMake'
+
+name = 'wannier90'
```
Comment (mboisson) [inline easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gofb-2020a.eb]: Needs to be named `Wannier90` (and the file too). The name must match the capitalization of the folder

File: easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gofb-2020a.eb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+version = '3.1.0'
+
+homepage = 'http://www.wannier.org/'
+description = "The Maximally-Localised Generalised Wannier Functions  calculates the counterpart of localized molecular orbitals for crystalline systems from the wavefuction."
+
+toolchain = {'name': 'gofb', 'version': '2020a'}
+toolchainopts = {'openmp': False}
+
+source_urls = ['https://github.com/wannier-developers/%(name)s/archive/']
```
Comment (mboisson) [inline easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gofb-2020a.eb]: will need to use `%(name_lower)s` if it is all in lower case

File: easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gofb-2020a.eb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+sources = ['v%(version)s.tar.gz']
+patches = ['%(name)s-%(version)s-gfort.patch']
```
Comment (mboisson) [inline easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gofb-2020a.eb]: need to adjust the name of the patch and use `%(name_lower)s`

File: easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gofb-2020a.eb [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+#postinstallcmds = ['mkdir %(installdir)s/bin %(installdir)s/lib && cp wannier90.x %(installdir)s/bin && cp libwannier.a %(installdir)s/lib']
+installopts = ['PREFIX=%(installdir)s install']
+
+parallel = 1
+
+skipsteps = ['configure']
+
+sanity_check_paths = {
+    'files': ['bin/%(name)s.x', 'bin/postw90.x', 'bin/w90chk2chk.x', 'bin/w90spn2spn.x', 'lib/libwannier.a'],
```
Comment (mboisson) [inline easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gofb-2020a.eb]: same, probably `%(name_lower)s` here

**Review by mboisson — CHANGES_REQUESTED** [review]
See comments in code

File: easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gfort.patch (hunk 1 above)
Comment (MoiseRousseau) [reply easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gfort.patch]: done

File: easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gfort.patch (hunk 2 above)
Comment (MoiseRousseau) [reply easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gfort.patch]: No reason. So I removed this change

File: easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gfort.patch (hunk 3 above)
Comment (MoiseRousseau) [reply easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gfort.patch]: done

File: easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gofb-2020a.eb (hunk 4 above)
Comment (MoiseRousseau) [reply easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gofb-2020a.eb]: done

File: easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gofb-2020a.eb (hunk 5 above)
Comment (MoiseRousseau) [reply easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gofb-2020a.eb]: done with `%(namelower)s`

File: easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gofb-2020a.eb (hunk 6 above)
Comment (MoiseRousseau) [reply easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gofb-2020a.eb]: done with `%(namelower)s`

File: easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gofb-2020a.eb (hunk 7 above)
Comment (MoiseRousseau) [reply easybuild/easyconfigs/w/Wannier90/wannier90-3.1.0-gofb-2020a.eb]: done with `%(namelower)s`

**Comment by mboisson** [comment]
lgtm

**Pull request merged**